### ISSUE TITLE
Loss:nan

### ISSUE THREAD

**Issue by XChuanLee** [meetshah1995/pytorch-semseg#86]
I run train.py with all default (fcn8s and voc dataset).
Here comes that _Loss: nan_ all the time.
And the val eval output is always
```
Overall Acc:     0.006510795767016046
Mean Acc :       0.047619047619047616
FreqW Acc :      4.239046151979406e-05
Mean IoU :       0.00031003789366743073
```
So where's the problem..
@_@

**Comment by meetshah1995** [comment]
The loss function implementation has been recently changed by @adam9500370 's latest PR, I haven't trained the FCN with the newer loss function. 

Can you share the loss log?

**Comment by adam9500370** [comment]
Hi, @IronDroid2088 .
I have tried the current repo code and trained with all the default settings `python train.py`.
Loss has never been NaN over 5 epochs.
The eval output seems normal after 5 epochs:
```
('FreqW Acc : \t', 0.7625165470098764)
('Overall Acc: \t', 0.8539931650734343)
('Mean Acc : \t', 0.626205958933612)
('Mean IoU : \t', 0.4714427117698269)
```
I use python 2.7.14 and install using `pip install -r requirements.txt`.
Can you provide more information about your training settings?

**Comment by BUPTAlanMa** [comment]
@adam9500370 hello， I trained UNet using PASCAL VOC 2012. I found the loss keep unchanged after serval epoch. And the output of valid always be 0. I cound not found why? Change the learning rate seem not work..

**Comment by adam9500370** [comment]
Hi, @BUPTAlanMa .
I have tried to train with `python train.py --arch unet --dataset pascal --img_rows 572 --img_cols 572`, and the loss changed every iteration in 3 epochs.
However, I observed that the model outputs will predict all 0 for every pixel after ~400 training iterations.

I think since proportion of pixels for segmentation label 0 (background) in pascal voc dataset is very high (class imbalance), you may need to do the followings:
1. UNet model weight initialization:
```
for m in model.modules():
    if isinstance(m, nn.Conv2d) or isinstance(m, nn.ConvTranspose2d):
        nn.init.kaiming_normal(m.weight, a=0, mode='fan_in', nonlinearity='relu')
        nn.init.constant(m.bias, 0.0)
```
2. Define different loss weight for class 0:
```
weight = torch.arange(n_classes).cuda()
weight[0] = 0.8 # lower loss weight for class 0
loss = loss_fn(input=outputs, target=labels, weight=weight)
```
After the above modifications, I think the model outputs can predict various classes.

**Comment by HPL123** [comment]
@adam9500370 hi，I have tried to train with python train.py --arch FCN8s --dataset camvid，the training process is wright. But the test result is very bad. I do not why
